AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Subscribe/DataStoreObserveQueryOperation.swift

```diff
@@ -230,6 +235,14 @@ class ObserveQueryTaskRunner<M: Model>: InternalTaskRunner, InternalTaskAsyncThr
                       receiveValue: self.onItemChangeAfterSync(mutationEvent:))
         }
     }
+    
+    func subscribeToModelSyncedEvent() {
+        readyEventSink = Amplify.Hub.publisher(for: .dataStore).sink { event in
+            if event.eventName == HubPayload.EventName.DataStore.modelSynced {
+                self.sendSnapshot()
+            }
```

Comment:
observeQuery requires a specific model type, such that all snapshots received are for one particular model. If developer wants to observe or observeQuery other models, they have to call the API with the other models, ie. `Amplify.DataStore.observeQuery(for: Post.self)`, `Amplify.DataStore.observeQuery(for: Blog.self)`, etc. 

The missing snapshot is when the synced state of the model toggles from false to true, which is why we are listening to the event here. This subscriber is listening to datastore events, checking if the event is "ModelSynced" event, and should additional filter on the modelType that this observeQuery is observing.

If we don't do this, if there are multiple models being synced, each one will cause the system to emit a modelSynced event, and each one will cause this observeQuery to send a snapshot. 

To filter on the modelName, we should be able to check the `event.payload` and cast to ModelSyncedEvent, and check that the modelSyncedEvent.modelName is equal to the modelSchema.name

Something like
```
if event.eventName == HubPayload.EventName.DataStore.modelSynced && let modelSyncedEvent == event.payload as? ModelSyncedEvent, modelSyncedEvent.modelName == modelSchema.name {
   self.sendSnapshot()
}
```